main.rb

```diff
@@ -0,0 +1,137 @@
+module Enumerable
+  # my_each method
+  def my_each
+    arr = to_a
+    return arr.to_enum unless block_given?
+
+    i = 0
+    while i < arr.size
+      yield arr[i]
+      i += 1
+    end
+  end
+
+  # my_each_with_index method
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  # my_select method
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    to_a.my_each { |i| arr << i if yield i }
+    arr
+  end
+
+  # my_all method
+  def my_all?(args = nil)
+    if block_given?
+      my_each { |i| return false unless yield i }
+      return true
+    elsif args.nil?
+      my_each { |i| return false if i.nil? }
+    elsif !args.nil? && args.is_a?(Class)
+      my_each { |i| return false unless i.class != args }
+    elsif !args.nil? && args.class == Regexp
+      my_each { |i| return false unless args.match(i) }
+    else
+      my_each { |i| return false if i != args }
+    end
+    true
+  end
+
+  # my_any method
+  def my_any?(args = nil)
+    if block_given?
+      my_each { |i| return true if yield i }
+      return false
+    elsif args.nil?
+      my_each { |i| return true if i }
+    elsif !args.nil? && args.is_a?(Class)
+      my_each { |i| return true if i.class != args }
+    elsif !args.nil? && args.class == Regexp
+      my_each { |i| return true if args.match(i) }
+    else
+      my_each { |i| return true if i != args }
+    end
+    false
+  end
+
+  # my_none method
+  def my_none?(args = nil)
+    if block_given?
+      my_each { |i| return false if yield i }
+      true
+    else
+      !my_any?(args)
+    end
+  end
+
+  # my_count method
+  def my_count(args = nil)
+    count = 0
+    if args.nil?
+      block_given? ? to_a.my_each { |i| count += 1 if yield(i) } : count = to_a.length
+    else
+      to_a.my_each { |i| count += 1 if args == i }
+    end
+    count
+  end
+
+  # my_map method
+  def my_map(proc = nil)
+    return to_enum unless block_given? || proc
+
+    arr = []
+    if proc
+      my_each { |i| arr << proc.call(i) }
+    else
+      my_each { |i| arr << yield(i) }
+    end
+    arr
+  end
+end
+
+module Injection
+  include Enumerable
+  # my_inject method
+  def my_inject(acc = 0)
+    result = 0
+    arr = self
+    arr = arr.to_a
+    if acc.is_a?(Symbol)
+      case acc
+      when :+
+        arr.my_each { |a| result += a }
+        result
+      when :-
+        arr.my_each { |a| result -= a }
+        result
+      when :*
+        result = 1
+        arr.my_each { |a| result *= a }
+        result
+      end
+    else
+      arr.my_each { |a| acc = yield(acc, a) }
+      acc
+    end
+  end
```

Comment:
No need to put it into that module **Injection** leave him into the same module than the others.
Kindly move it as the others.

Reply:
```rb
2.6.5 :058 > search = proc { |memo, word| memo.length > word.length ? memo : word }
2.6.5 :059 > words.my_inject(&search)
Traceback (most recent call last):
       10: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        9: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        8: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        5: from (irb):59
        4: from main.rb:123:in `my_inject'
        3: from main.rb:9:in `my_each'
        2: from main.rb:123:in `block in my_inject'
        1: from (irb):58:in `block in irb_binding'
NoMethodError (undefined method `length' for 0:Integer)
2.6.5 :060 > words.inject(&search)
 => "blade" 
2.6.5 :061 > 
```

Reply:
```rb
2.6.5 :063 > range = Range.new(5, 50)
2.6.5 :064 > range.my_inject(2, :*)
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        3: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        2: from (irb):64
        1: from main.rb:105:in `my_inject'
ArgumentError (wrong number of arguments (given 2, expected 0..1))
2.6.5 :065 > range.inject(2, :*)
 => 2534507766809448170301050680505397403698136797413376000000000000 
2.6.5 :066 >
```

Reply:
Thank you for the review.
The problem is that when I put everything in the same module, I get an error that the module has many lines.